### PR TITLE
comparison of a muni to similar munis

### PR DESCRIPTION
**Pull request by jbothma**
## MIGRATIONS: 
- Run locally against prod DB before deploying code
- Run 0002 with --fake because the tables already exist
- Run the rest for real.

## New data:
- use the new sql/import_development_code.sql to import the attached pdg_muni_classification.txt (fixing path and extension set to txt for github)

## Changes
- start managing SQL using django models
- add muni development code
- remove prodile_data dependency on django
- add script to export json and csv views of scorecard indicator data

### CODE REVIEW

**Comment by longhotsummer** [comment]
You're going to need to dump the updated scorecard_geography table so that it contains the development code column.

**Comment by jbothma** [comment]
Woops - should add this to the PR: use the new sql/import_development_code.sql and the attached file (fixing path and extension set to txt for github)
[pdg_muni_classification.txt](https://github.com/Code4SA/municipal-data/files/611465/pdg_muni_classification.txt)

**Comment by longhotsummer** [comment]
Can't you just dump the updated table? Will be easier when starting from an empty db, too.

**Comment by jbothma** [comment]
sure

**Comment by jbothma** [comment]
@longhotsummer https://drive.google.com/open?id=0B8yXDspOrgF9bGNyTS16RDYyVEE

**Comment by longhotsummer** [comment]
FYI, We're missing medians for the 'A' category. The data sheet they sent didn't have metros, we need to set in the DB (currently it's null).

**Comment by longhotsummer** [comment]
I've merged in my comparatives so that we're both up to date. It's looking good. Ignore the provincial values in the screenshot, they're random.

umzimvubu - municipal money 2016-11-24 18-13-38

File: bin/indicator_distributions.py [hunk 1]
```diff
@@ -0,0 +1,118 @@
+import sys
+sys.path.append('.')
+
+import numpy as np
```
Comment (jbothma) [inline bin/indicator_distributions.py]: remove unused numpy

File: bin/indicator_distributions.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+from scorecard.profile_data import MuniApiClient
+import argparse
+import json
+from indicators import get_munis
+from collections import defaultdict
```
Comment (jbothma) [inline bin/indicator_distributions.py]: sort imports

File: bin/indicator_distributions.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    'current_ratio',
+    'liquidity_ratio',
+    'op_budget_diff',
+    'rep_maint_perc_ppe',
+    'wasteful_exp'
+]
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Tool to dump the materialised views of the municipal finance data used on the Municipal Money website.')
```
Comment (jbothma) [inline bin/indicator_distributions.py]: duplicates bin/indicators.py

File: municipal_finance/models.py [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+    label = models.TextField(blank=True, null=True)
+    position_in_return_form = models.IntegerField(blank=True, null=True)
+    return_form_structure = models.TextField(blank=True, null=True)
+    composition = models.TextField(blank=True, null=True)
+
+    class Meta:
+        db_table = 'repmaint_items'
+
+
+class ScorecardGeography(models.Model):
```
Comment (jbothma) [inline municipal_finance/models.py]: remove repeat of scorecard.geo

File: requirements.txt [hunk 5]
```diff
@@ -55,3 +55,4 @@ wheel==0.24.0
 whitenoise==1.0.6
 xlrd==0.9.4
 xlsxwriter==0.9.2
+numpy==1.11.2
```
Comment (jbothma) [inline requirements.txt]: remove unused

File: scorecard/profiles.py [hunk 6]
```diff
@@ -1,8 +1,63 @@
...
+    'current_ratio',
+    'liquidity_ratio',
+    'op_budget_diff',
+    'rep_maint_perc_ppe',
+    'wasteful_exp',
+]
+
+
+def build_comparisons(geo, indicators, medians):
+    # TODO: move the indicator metadata into the indicator calculator object itself
```
Comment (jbothma) [inline scorecard/profiles.py]: https://www.pivotaltracker.com/story/show/134953197

**Comment by jbothma** [comment]
some benefits I'm experiencing of storing the materialised views in files instead of a DB are
- git diff can show me whether refactoring changed calculation unintentionally
- it's about as much effort as writing code to store in DB but we know we'd like to be less-dependent on the DB for profiles

**Comment by jbothma** [comment]
this is a bit confusing - I read this as 24% is 25% higher than 17% ie 17 + 25 = 42% while really it meant 24% is about (25% of 17 = 4.25) bigger = close to 22.5%

we said we should perhaps not use percent when talking about percent

percent-of-percent-muni-comparison

File: requirements.txt (hunk 5 above)
Comment (jbothma) [inline requirements.txt]: :+1:

File: bin/indicator_distributions.py (hunk 3 above)
Comment (jbothma) [inline bin/indicator_distributions.py]: fixed

File: bin/indicator_distributions.py (hunk 2 above)
Comment (jbothma) [inline bin/indicator_distributions.py]: fixed

**Comment by jbothma** [comment]
@longhotsummer please run `delete from django_migrations where name in ('0003_remove_geography_osm_area_id', '0003_scorecardgeography_development_category', '0004_delete_scorecardgeography');`

I deleted the unnecessary migrations from this branch instead. what's te worst that can happen?

**Comment by jbothma** [comment]
Confirm calculation of median of over/under spending

File: municipal_finance/models.py (hunk 4 above)
Comment (jbothma) [inline municipal_finance/models.py]: fixed

**Comment by jbothma** [comment]
I'm pretty happy with this branch for deployment now. happy for more feedback and the things we want to check with Treasury.

**Comment by longhotsummer** [comment]
This LGTM. As discussed, happy to merge this when you are. We'll follow up with https://github.com/Code4SA/municipal-data/pull/340

**Pull request merged**